core/src/main/java/com/datastax/oss/driver/internal/core/DefaultCluster.java

```diff
@@ -145,25 +149,63 @@ private void init() {
           .addContactPoints(initialContactPoints)
           .thenCompose(v -> context.topologyMonitor().init())
           .thenCompose(v -> metadataManager.refreshNodes())
-          .thenAccept(
-              v -> {
-                try {
-                  context.loadBalancingPolicyWrapper().init();
-                  context.configLoader().onDriverInit(context);
-                  LOG.debug("[{}] Initialization complete, ready", logPrefix);
-                  initFuture.complete(DefaultCluster.this);
-                  // TODO schedule full schema refresh asynchronously (does not block init)
-                } catch (Throwable throwable) {
-                  initFuture.completeExceptionally(throwable);
-                }
-              })
+          .thenAccept(this::afterInitialNodeListRefresh)
           .exceptionally(
               error -> {
                 initFuture.completeExceptionally(error);
                 return null;
               });
     }
 
+    private void afterInitialNodeListRefresh(@SuppressWarnings("unused") Void ignored) {
+      try {
+        boolean protocolWasForced =
+            context.config().getDefaultProfile().isDefined(CoreDriverOption.PROTOCOL_VERSION);
+        boolean needSchemaRefresh = true;
+        if (!protocolWasForced) {
+          ProtocolVersion currentVersion = context.protocolVersion();
+          ProtocolVersion bestVersion =
+              context
+                  .protocolVersionRegistry()
+                  .highestCommon(metadataManager.getMetadata().getNodes().values());
```

Comment:
It would be great in theory, but the problem is that the LBP has not been initialized yet, so we don't know distances.
So it sounds reasonable to file that as an edge case where you would have to force the version.